src/directory/read_only_source.rs

```diff
@@ -42,7 +42,7 @@ impl ReadOnlySource {
     pub fn as_slice(&self) -> &[u8] {
         match *self {
             #[cfg(feature = "mmap")]
-            ReadOnlySource::Mmap(ref mmap_read_only) => unsafe { mmap_read_only.as_slice() },
+            ReadOnlySource::Mmap(ref mmap_read_only) => mmap_read_only.as_slice(),
```

Comment:
I don't understand why this change was required when I updated fst -> 0.3 (could be new rust stable too) - but would want someone else's eyes on it.